arwen/elrondapi/elrondei.go

```diff
@@ -1939,6 +2062,45 @@ func executeOnDestContext(
 		return 1
 	}
 
+	if host.IsESDTFunctionsEnabled() {
+		function, err := runtime.MemLoad(functionOffset, functionLength)
```

Comment:
Please create the function `host.detectSyncESDTTransfer(functionOffset, functionLength, addressOffset)` returning `function, destination, isESDTTransfer, err` to simplify this code.

Reply:
done